src/System.Runtime/tests/System/GCTests.netstandard1.7.cs

```diff
@@ -0,0 +1,195 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics;
+using System.Collections.Generic;
+using System.Threading;
+using System.Runtime;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.Tests
+{
+    public class GCExtendedTests : RemoteExecutorTestBase
+    {
+        private const int TimeoutMilliseconds = 10*30 * 1000; //if full GC is triggered it may take a while
+      
+        [Fact]
+        public static void GetGeneration_WeakReference()
+        {
+            RemoteInvokeOptions options = new RemoteInvokeOptions();
+            options.TimeOut = TimeoutMilliseconds;
+            RemoteInvoke(() =>
+                {
+
+                    Func<WeakReference> getweakref = delegate ()
+                    {
+                        Version myobj = new Version();
+                        var wkref = new WeakReference(myobj);
+
+                        Assert.True(GC.TryStartNoGCRegion(1024));
```

Comment:
The same reason i expect other tests that assert the same condition
That process is unlikely to allocate more than 1Kb( actually the hope is it does not allocate at all) in the no gc region.

Am i missing something here ?